integration-tests/src/test/java/org/apache/fineract/integrationtests/RepaymentReverseExternalIdTest.java

```diff
@@ -0,0 +1,148 @@
+            globalConfigurationHelper.updateGlobalConfiguration(GlobalConfigurationConstants.ENABLE_BUSINESS_DATE,
+                    new PutGlobalConfigurationsRequest().enabled(true));
+            BusinessDateHelper.updateBusinessDate(requestSpec, responseSpec, BusinessDateType.BUSINESS_DATE, todaysDate);
+
+            // Create a client and a loan with externalId
+            final Integer clientId = clientHelper.createClient(ClientHelper.defaultClientCreationRequest()).getClientId().intValue();
+            final GetLoanProductsProductIdResponse loanProduct = createLoanProduct(loanTransactionHelper, null);
+            final String loanExternalId = UUID.randomUUID().toString();
+            final Integer loanId = createLoanAccount(clientId, loanProduct.getId(), loanExternalId);
+
+            // First repayment to cover part of the loan
+            final PostLoansLoanIdTransactionsResponse repaymentTransaction1 = loanTransactionHelper.makeLoanRepayment(loanExternalId,
+                    new PostLoansLoanIdTransactionsRequest().dateFormat("dd MMMM yyyy").transactionDate(firstRepaymentDate).locale("en")
+                            .transactionAmount(firstRepaymentAmount));
+
+            // Second repayment to create overpayment
+            final PostLoansLoanIdTransactionsResponse repaymentTransaction2 = loanTransactionHelper.makeLoanRepayment(loanExternalId,
+                    new PostLoansLoanIdTransactionsRequest().dateFormat("dd MMMM yyyy").transactionDate(secondRepaymentDate).locale("en")
+                            .transactionAmount(secondRepaymentAmount)); // This creates an overpayment as total is now
+                                                                        // 1010
+
+            // Verify that the loan is marked as overpaid
+            GetLoansLoanIdResponse loanDetailsOverpaid = loanTransactionHelper.getLoanDetails((long) loanId);
+            assertTrue(loanDetailsOverpaid.getStatus().getOverpaid()); // Overpaid status should be true
+            assertNotNull(loanDetailsOverpaid.getOverpaidOnDate()); // Overpaid date should be set
+            assertEquals(loanDetailsOverpaid.getExternalId(), loanExternalId); // externalId should match the original
+
+            // Reverse the second repayment to remove the overpayment
+            loanTransactionHelper.reverseRepayment(loanId, repaymentTransaction2.getResourceId().intValue(), reverseDate);
+
+            // Verify that the loan is no longer overpaid and overpaid date is reset
+            GetLoansLoanIdResponse loanDetailsNotOverpaid = loanTransactionHelper.getLoanDetails((long) loanId);
+            assertFalse(loanDetailsNotOverpaid.getStatus().getOverpaid()); // Overpaid status should be false
+            assertNull(loanDetailsNotOverpaid.getOverpaidOnDate()); // Overpaid date should be reset
+            assertEquals(loanDetailsNotOverpaid.getExternalId(), loanExternalId); // externalId should remain unchanged
```

Comment:
Please make sure you are validating the reversed transaction external id is not changed! We don't need to check whether the external id of the loan got changed!